tools/rpm2img

```diff
@@ -320,20 +340,21 @@ fi
 cat <<EOF > "${BOOT_MOUNT}/grub/grub.cfg"
 set default="0"
 set timeout="0"
+set dm_verity_root="${DM_VERITY_ROOT[@]}"
 
 menuentry "${PRETTY_NAME} ${VERSION_ID}" {
    linux (\$root)/vmlinuz \\
        ${KERNEL_PARAMETERS} \\
        ${BOOTCONFIG} \\
        root=/dev/dm-0 rootwait ro \\
        raid=noautodetect \\
-       random.trust_cpu=on selinux=1 enforcing=1 \\
-       dm_verity.max_bios=-1 dm_verity.dev_wait=1 \\
-       dm-mod.create="root,,,ro,0 $VERITY_DATA_512B_BLOCKS verity $VERITY_VERSION PARTUUID=\$boot_uuid/PARTNROFF=1 PARTUUID=\$boot_uuid/PARTNROFF=2 \\
-       $VERITY_DATA_BLOCK_SIZE $VERITY_HASH_BLOCK_SIZE $VERITY_DATA_4K_BLOCKS 1 $VERITY_HASH_ALGORITHM $VERITY_ROOT_HASH $VERITY_SALT \\
-       2 restart_on_corruption ignore_zero_blocks" \\
+       random.trust_cpu=on \\
+       selinux=1 enforcing \\
```

Comment:
From my reading of the code, `enforcing` without any value won't actually be recognized as a valid option. On closer look, the current kernels don't even have the respective config options for these two command line options enabled (`CONFIG_SECURITY_SELINUX_BOOTPARAM` for `selinux` and `CONFIG_SECURITY_SELINUX_DEVELOP` for `enforcing`). This is fine in so far as both of them fall back to secure (enabled, enforcing) defaults.

It would need some digging to find out when the config options were dropped. Should they be brought back? Being unable to disable SELinux via boot-time options might not fundamentally change the security posture, but adds an extra hurdle.

Reply:
The kernels are configured without these options by design - SELinux isn't ever meant to be disabled on Bottlerocket.

I've retained `selinux=1 enforcing` on the off-chance that there's some application out there that tries to detect SELinux state by checking `/proc/cmdline`. That's pretty broken, but the correct approach - linking to libselinux - might not be appealing for various reasons. It serves as a bit of a hint - "here be SELinux". I'm not sure that hint is good for much other than reporting somewhere - "yup, this node probably has SELinux turned on".

So I've occasionally been tempted to drop them. It's a bit ugly that they get passed down to `init` as parameters, since the kernel doesn't recognize them. The safest way to do that would be to shift them into kernel parameters in all the existing variant definitions, and then omit them in newer variants going forward.

Reply:
Thanks for clarifying! I figured something like this could also have been the reason. Should you add the `=1` back onto `enforcing`, as right now it's probably not serving any purpose and wouldn't even be recognized if the kernel config option was enabled? Approving in any case, as the original doubts have been settled.

Reply:
> Should you add the =1 back onto enforcing, as right now it's probably not serving any purpose and wouldn't even be recognized if the kernel config option was enabled?

Oops - yes!